set.mm

```diff
@@ -730651,6 +730957,1157 @@ last digit of the integer part (for ` b = ` 10 the first digit before
+      GVDAHGDWHWIWLVCVDAWDXLXJAGEWIWKTAHDWHWLTAGDWIWLTVJAXLWEWDPAHDWHWLVGVHVIVH
+      VIVKVL $.
+  $}
+
+  ${
+    affineid.f $e |- ( ph -> A e. CC ) $.
+    affineid.x $e |- ( ph -> T e. CC ) $.
+    $( Identity of an affine combination.  (Contributed by AV, 2-Feb-2023.) $)
+    affineid $p |- ( ph -> ( ( ( 1 - T ) x. A ) + ( T x. A ) ) = A ) $=
+      ( c1 cmin co cmul caddc 1cnd subdird mulid2d oveq1d eqtrd mulcld npcand )
+      AFCGHBIHZCBIHZJHBSGHZSJHBARTSJARFBIHZSGHTAFCBAKEDLAUABSGABDMNONABSDACBEDP
+      QO $.
+  $}
+
+  ${
+    resum2sqcl.q $e |- Q = ( ( A ^ 2 ) + ( B ^ 2 ) ) $.
+    $( The sum of two squares of real numbers is a real number.  (Contributed
+       by AV, 7-Feb-2023.) $)
+    resum2sqcl $p |- ( ( A e. RR /\ B e. RR ) -> Q e. RR ) $=
+      ( cr wcel wa c2 cexp co caddc simpl resqcld simpr readdcld syl5eqel ) AEF
+      ZBEFZGZCAHIJZBHIJZKJEDSTUASAQRLMSBQRNMOP $.
+
+    $( The sum of the square of a nonzero real number and the square of another
+       real number is greater than zero.  (Contributed by AV, 7-Feb-2023.) $)
+    resum2sqgt0 $p |- ( ( ( A e. RR /\ A =/= 0 ) /\ B e. RR ) -> 0 < Q ) $=
+      ( cr wcel cc0 wne wa c2 co caddc clt simpl resqcld adantr simpr wbr sqgt0
+      cexp cle sqge0 adantl addgtge0d syl6breqr ) AEFZAGHZIZBEFZIZGAJTKZBJTKZLK
+      CMUJUKULUHUKEFUIUHAUFUGNOPUJBUHUIQOUHGUKMRUIASPUIGULUARUHBUBUCUDDUE $.
+  $}
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Spheres and lines in real Euclidean spaces
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c LineM Sphere $.
+
+  $( Declare the syntax for lines in generalized real Euclidean spaces. $)
+  cline $a class LineM $.
+
+  $( Declare the syntax for spheres in generalized real Euclidean spaces. $)
+  csph $a class Sphere $.
+
+  ${
+    $d p t w x y $.
+    $( Definition of lines passing through two different points in a left
+       module (or any extended structure having a base set, an addition, and a
+       scalar multiplication).  (Contributed by AV, 14-Jan-2023.) $)
+    df-line $a |- LineM = ( w e. _V |-> ( x e. ( Base ` w ) ,
+                                         y e. ( ( Base ` w ) \ { x } )
+                    |-> { p e. ( Base ` w ) | E. t e. ( Base ` ( Scalar ` w ) )
+                          p = ( ( ( ( 1r ` ( Scalar ` w ) )
+                                    ( -g ` ( Scalar ` w ) ) t ) ( .s ` w ) x )
+                                ( +g ` w ) ( t ( .s ` w ) y ) ) } ) ) $.
+  $}
+
+  ${
+    $d p r w x y $.
+    $( Definition of spheres for given centers and radii in a metric space (or
+       more general, in a distance space, see ~ distspace , or eben in any
+       extended structure having a base set and a distance function into the
+       real numbers.  (Contributed by AV, 14-Jan-2023.) $)
+    df-sph $a |- Sphere = ( w e. _V |-> ( x e. ( Base ` w ) ,
+                                          r e. ( 0 [,] +oo )
```

Comment:
I would use `r e. RR*` since this could make some future uses easier.

Reply:
Is there actually a meaningful application for a sphere with negativ radius? If not, is it really easier to cope with `` ( M ( Sphere ` E ) R ) = (/) `` if ` R e. RR* ` and ` R < 0 ` instead of assuming ` 0 <_ R ` explicitly?

Reply:
Indeed: a distance has values in $[0,\infty]$ so why define a sphere of negative radius anymore than of radius equal to any random set ? The reason is that you can obtain a negative number by subtracting two positive ones, so defining "spheres" of negative radius (as the empty set, obviously) might allow one to dispense of special degenerate cases in proofs. This is only a matter of convenience.

I do not have a specific example right now, but for instance in RR^3, the cone $\{ (x, y, z) \in \mathbb{R}^3 \mid \sqrt{x^2 + y ^2} = z \}$ intersects each plane z=cte in a "sphere" (here a circle) of radius z, so the formal version of that would not have a special case for z<0.

Also, "undefined values" are currently set to (/) so it's true that it wouldn't change the value here, but explicitly defining it as (/) is cleaner since it does not depend on an outside convention.

Reply:
As long as negative radii  are not really required I would leave this as it is.

Reply:
Your mathbox, your choice.